### PR TITLE
Use apps/v1 ReplicaSet in controller and tests.

### PR DESCRIPTION
**Pull request by enisoc**
This updates the RS/RC controller and RS integration/e2e tests to use apps/v1 ReplicaSet, as part of #55714.

It does *not* update the Deployment controller, nor its integration/e2e tests, to use apps/v1 ReplicaSet. That will be done in a separate PR (#61419) because Deployment has many more tendrils embedded throughout the system.

```release-note
Conformance: ReplicaSet must be supported in the `apps/v1` version.
```

/assign @janetkuo

### CODE REVIEW

**Review by janetkuo — COMMENTED** [review]
Comment on the last commit: `ReplicaSet: Remove integration tests for legacy behavior.`

IMO those tests still have their value. Legacy API versions, although deprecated, still need to function. The tests make sure they're not broken. I don't feel strongly about keeping them though. 

Other changes LGTM.

**Comment by enisoc** [comment]
@janetkuo Good point. I brought back the legacy tests and (hopefully) fixed them instead of removing them.

**Comment by enisoc** [comment]
/assign @deads2k 
for kube-controller-manager OWNERS

Tests are green again.

**Review by janetkuo — APPROVED** [review]

File: pkg/controller/replication/conversion.go [hunk 1]
```diff
@@ -315,23 +275,23 @@ func convertCall(fn func(*v1.ReplicationController) (*v1.ReplicationController,
...
 	var rsInternal extensions.ReplicaSet
 	if err := apiv1.Convert_v1_ReplicationController_to_extensions_ReplicaSet(rc, &rsInternal, nil); err != nil {
 		return nil, fmt.Errorf("can't convert ReplicationController %v/%v to ReplicaSet: %v", rc.Namespace, rc.Name, err)
 	}
 	if out == nil {
-		out = new(extensionsv1beta1.ReplicaSet)
+		out = new(apps.ReplicaSet)
 	}
-	if err := extensionsinternalv1beta1.Convert_extensions_ReplicaSet_To_v1beta1_ReplicaSet(&rsInternal, out, nil); err != nil {
+	if err := appsconversion.Convert_extensions_ReplicaSet_To_v1_ReplicaSet(&rsInternal, out, nil); err != nil {
```
Comment (fisherxu) [inline pkg/controller/replication/conversion.go]: Whether we can try convert RC to RS don't through extension? :-)

**Comment by deads2k** [comment]
approving for `cmd/kube-controller-manager`  changes

/approve

File: pkg/controller/replication/conversion.go (hunk 1 above)
Comment (enisoc) [reply pkg/controller/replication/conversion.go]: @fisherxu Do you have a concern about the correctness of the code? Or is this just a comment on the fact that this looks weird?

For context on why it looks weird: The internal (unversioned) [ReplicaSet struct](https://github.com/kubernetes/kubernetes/blob/bcf517e37251cf41ecb7e1fd1f5fd23a7458fcf6/pkg/apis/extensions/types.go#L679-L692) is still in `extensions`. So to convert between `apps/v1` ReplicaSet and `core/v1` ReplicationContorller, we go via the unversioned internal `extensions` ReplicaSet.

Comment (fisherxu) [reply pkg/controller/replication/conversion.go]: In fact, I have tried to bump deployment/replicaset to apps/v1 last week, but haven't dealt with the rollback problem yet :-( and so not send a PR. Then observed your PR, so great~
To this conversion, I also through  extensions in my local env. I was thinking if we can move internal RS/Deployment to apps in a follow-up PR?

Comment (fisherxu) [reply pkg/controller/replication/conversion.go]: @enisoc If we need move internal RS/Deployment to apps, I can help do it in a follow-up PR :)

Comment (enisoc) [reply pkg/controller/replication/conversion.go]: @fisherxu Sorry about overlapping with your work. Did you post somewhere that you were working on that? FYI I started Deployment in #61419 but that's currently blocked by #61433.

I agree we should move the internal struct to `apps` in a separate PR. If you plan to work on that, please leave a comment on #55714 so we don't overlap. Thanks!

Comment (fisherxu) [reply pkg/controller/replication/conversion.go]: @enisoc Has sent a PR #61755 to move the internal struct to apps, and working on the failed CI :)

**Pull request merged**